helpers_test.go

```diff
@@ -50,3 +52,20 @@ func formatUnequalValues(got, want interface{}) string {
 
 	return fmt.Sprintf("\ngot: %s\nwant: %s", a, b)
 }
+
+func assertBaggageStringsEqual(t *testing.T, got, want string, userMessage ...interface{}) {
+	t.Helper()
+
+	baggageGot, err := baggage.Parse(got)
+	if err != nil {
+		t.Error(err)
+	}
+	baggageWant, err := baggage.Parse(want)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !reflect.DeepEqual(baggageGot, baggageWant) {
```

Comment:
I would always use https://pkg.go.dev/github.com/google/go-cmp/cmp, had quite a few issues with reflect in the past.

Reply:
Agreed, will include things like this in the planned test utils refactoring.